onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -65,169 +64,217 @@ def onnx_dtype(self):
+        )
+    if isinstance(value, TorchScriptTensor):
+        return value.symbolic_value()
     # A normal python value
     return value
 
 
 def _wrap_torch_value_to_tensor(
-    value: Union[Dict[str, Any], List]
-) -> Union[Dict[str, Any], List]:
-    # wrap torch.Value with TorchScriptTensor
+    value: torch.Value
+    | dict[str, torch.Value | str | int | float]
+    | list[torch.Value | str | int | float]
+    | tuple[torch.Value | str | int | float, ...]
+) -> ValidArgumentType | dict[str, ValidArgumentType] | list[ValidArgumentType] | tuple[
+    ValidArgumentType, ...
+]:
+    """Wrap torch.Value to TorchScriptTensor.
+
+    Args:
+        value: The value to wrap.
+
+    Returns:
+        The wrapped value.
+    """
     if isinstance(value, dict):
-        value = {
+        return {
             k: TorchScriptTensor(v) if isinstance(v, torch.Value) else v
             for k, v in value.items()
         }
-    elif isinstance(value, list):
-        value = [TorchScriptTensor(v) if isinstance(v, torch.Value) else v for v in value]
-    elif isinstance(value, tuple):
+    if isinstance(value, list):
+        return [TorchScriptTensor(v) if isinstance(v, torch.Value) else v for v in value]
+    if isinstance(value, tuple):
         return tuple(TorchScriptTensor(v) if isinstance(v, torch.Value) else v for v in value)
-    elif isinstance(value, torch.Value):
-        value = TorchScriptTensor(value)
+    if isinstance(value, torch.Value):
+        return TorchScriptTensor(value)
     return value
 
+
 def _unwrap_tensors_to_torch_values(tensors):
+    # TODO(justinchuby): Do we really need this?
     if isinstance(tensors, Sequence):
         return [_unwrap_tensor_to_torch_value(output) for output in tensors]
     return _unwrap_tensor_to_torch_value(tensors)
 
+
 class TorchScriptEvaluator(evaluator.Evaluator):
+    """An onnxscript Evaluator that captures the graph into torchscript."""
+
     def __init__(self, graph: TorchScriptGraph):
-        self._graph = graph
+        self._graph: TorchScriptGraph = graph
 
     @property
     def graph(self) -> TorchScriptGraph:
         return self._graph
 
-    def eval_function(self, function: onnxscript.OnnxFunction, args, kwargs):
+    def eval_function(
+        self,
+        function: onnxscript.OnnxFunction,
+        args: Sequence[ValidArgumentType],
+        kwargs: Sequence[ValidArgumentType],
+    ):
         # args/kwargs are TorchScriptTensor/python built-in based
         inputs, attributes = _split_args_kwargs_to_input_attr(function, args, kwargs)
         return self._graph.add_function(function, inputs, attributes)
 
-    def _eval(self, schema, inputs, attributes, closure):
-        # TODO: Does it really know what the inputs are?
+    def _eval(self, schema: onnx.defs.OpSchema, inputs, attributes, closure: Any):
+        # TODO(justinchuby): Does it really know what the inputs are?
```

Comment:
IsOpschema info sufficient for us to split input/attr like we did in ONNXfunction?

Reply:
I think so. Will need to validate